### ISSUE TITLE
BT stops working if Wifi connected on RTL8723BE

### ISSUE THREAD

**Issue by freechelmi** [lwfinger/rtl8723au_bt#19]
On RTL8723BE (13d3:3410 )  Mini PCIe (Brix N2807) BT works great with the new branch, but stops connecting to devices if wifi is connected to a hotspot. btcoexist is loaded.

Will give details later on.

**Comment by chzw517** [comment]
i have the same issue... anyone can help?

**Comment by lwfinger** [comment]
What driver are you using for the RTL8723BE? There are some fixes for BT coexistence that are in the process of being submitted for kernel 3.20. You can use those by building driver rtl8723be from the 'troy' branch of the repo at http://github.com/lwfinger/rtlwifi_new.git.

**Comment by sunny0** [comment]
Just tried troy's branch with 3.18.3 and kernel panic.

**Comment by lwfinger** [comment]
Pl.ease post a photo of the panic. Otherwise I cannot help.

**Comment by sunny0** [comment]
How's this?
![img_20150123_231744](https://cloud.githubusercontent.com/assets/10410996/5886143/78196ad0-a359-11e4-919b-0aa306fc7638.jpg)

**Comment by chzw517** [comment]
It causes a kernel panic!....I using ubuntu15.04 alpha1 with kernel 3.18.0...and...then...nothing left...:'(

**Comment by lwfinger** [comment]
The panic should be fixed now.

**Comment by chzw517** [comment]
Can you tell us what was the problem ?and How can you fix that?

**Comment by lwfinger** [comment]
It was a coding mistake. If you want the details, then do 'git show caa1026'.

**Comment by sunny0** [comment]
Ok, no more panic, but still no devices shown in bluetoothctl. rfkill shows no blocks too...

**Comment by chzw517** [comment]
Hi..sunny0...pls keep us informed if you solved the problem...thank you very much...

**Comment by ohgodwhyyy** [comment]
Same problem with last two Ubuntu 14.10 kernels (3.16.0-29 and 3.16.0-30).
I had to do modprobe -r rtl8723be in order to get bluetooth to work but then wifi is gone...

**Comment by lwfinger** [comment]
FYI, there is a new 'kernel' branch that has code that will be in kernel 3.21. It handles the Realtek devices just like any other kernel device. Thus far, no changes for older kernels have been included. As problems are reported, I will fix them.

**Comment by sunny0** [comment]
Not sure where to reply anymore... haha

Anyway, I tried out the latest wifi and bluetooth drivers against 3.18.3 on opensuse from the newly made kernel branches. Still no devices listed in bluetoothctl, no blocks, nothing in dmesg. Wifi works fine, bluetooth is invisible...

**Comment by lwfinger** [comment]
Please check the output of the dmesg command. Anything show up there? In particular, did you 'sudo make install' so that the firmware is availablle.

**Comment by sunny0** [comment]
Yes, i did 'sudo make install' after 'make'. Checked dmesg and nothing new is shown.

**Comment by sunny0** [comment]
Reverted back to the 3.18.3 firmware and modules, did 'sudo make install' for just the bluetooth kernel branch, then 'sudo modprobe -rv btusb', and 'sudo modprobe -v btusb', and this is dmesg:

usbcore: deregistering interface driver btusb
Bluetooth: hci0 urb ffff880436678700 failed to resubmit (2)
usbcore: registered new interface driver btusb
Bluetooth: hci0: hci_ver=06 hci_rev=1e3e lmp_ver=06 lmp_subver=4ce1
Bluetooth: rtl8723b: assuming no firmware upload needed.

**Comment by lwfinger** [comment]
Your posting is confusing. First it did not load BT firmware, then it did.

It appears that your device needs code changes. I add some code to handle that subversion. Do a 'git pull' on the kernel branch and try again to 'make', 'sudo make install', 'sudo modprobe -rv btusb', and 'sudo modprove -v install'.

**Comment by sunny0** [comment]
That fixed it!

**Comment by lwfinger** [comment]
Please post the full output of the dmesg command.

**Comment by chzw517** [comment]
Hi lwfinger... what branch should I pull? Is it master?

**Comment by lwfinger** [comment]
I suggest the kernel branch.

**Comment by freechelmi** [comment]
Hi Larry , sirry I was not really responding on this point for few weeks.

We want to work on Ubuntu 3.16 Kernel , What branch for Wifi & BT do you recommand for rtl8723be ? 

I 'm a bit lost here because it seems that the 'kernel' branch might not work on older kernel ?

I will try the kernel branch anyway.

**Comment by lwfinger** [comment]
The best BT code is in the kernel branch. It will work for any kernel 3.14 or newer.

The best wifi code is in http://github.com/lwfinger/rtlwifi_new.git using the master or troy branch.

**Comment by freechelmi** [comment]
Thanks larry. your work is really valuable.

Will try to produce dkms deb as before. as I get this now should work with Wifi+BT at the same time which is great.

**Comment by freechelmi** [comment]
Hi , I tried the kernel branch on lastest Ubuntu 14.04.2 (kernel 3.16.X)  , and even if the firmware seems correctly loaded the result is the same as the mainline btusb : no discovering anything. anybody experienced this ?

**Comment by Akhenaton** [comment]
Hi,
Yes, same here !  :-)

**Comment by lwfinger** [comment]
The *only* sure way to get this to work is to use the kernel branch of this repo. The first kernel with this version of the driver will be 4.1-rc1, which will be out in about one month.

**Comment by freechelmi** [comment]
Yes we all use the kernel branch as I said.

**Comment by lwfinger** [comment]
Sorry, I misread your comment.

@freechelmi: I see you have RTL8723BE hardware. Your device is in the USB tables, and it should be recognized.

Do you have the latest version, which includes commit 47629cd82ce0? That fixed a typo that prevented the firmware from being installed correctly.

If you have that commit, please post the section of the output of dmesg where the device is first recognized.

**Comment by soyzenitram** [comment]
Same issue (bt stops when wifi is on):
Kernel version 4.4.0-45
Wifi driver: rtlwifi_new, rock.new_btcoex branch, 
BT driver: this one, kernel branch

Any help please???

Read something about making a git pull, I don't know if it is suitable for my situation, not very familiarized with git commands and their effects... sorry.
